src/config/mod.rs

```diff
@@ -179,79 +152,56 @@ impl Config {
             .expect("Failed to add cosmic-config to the event loop");
         let xdg = xdg::BaseDirectories::new().ok();
         let workspace = get_config::<WorkspaceConfig>(&config, "workspaces");
+
         let cosmic_comp_config =
             CosmicCompConfig::get_entry(&config).unwrap_or_else(|(errs, c)| {
                 for err in errs {
                     error!(?err, "");
                 }
                 c
             });
-        Config {
-            static_conf: Self::load_static(xdg.as_ref(), workspace.workspace_layout),
-            dynamic_conf: Self::load_dynamic(xdg.as_ref()),
-            cosmic_conf: cosmic_comp_config,
-            cosmic_helper: config,
-        }
-    }
-
-    fn load_static(
-        xdg: Option<&xdg::BaseDirectories>,
-        workspace_layout: WorkspaceLayout,
-    ) -> StaticConfig {
-        let mut locations = if let Some(base) = xdg {
-            vec![
-                base.get_config_file("cosmic-comp.ron"),
-                base.get_config_file("cosmic-comp/config.ron"),
-            ]
-        } else {
-            Vec::with_capacity(3)
-        };
-        if cfg!(debug_assertions) {
-            if let Ok(mut cwd) = std::env::current_dir() {
-                cwd.push("config.ron");
-                locations.push(cwd);
-            }
-        }
-        locations.push(PathBuf::from("/etc/cosmic-comp/config.ron"));
-        locations.push(PathBuf::from("/etc/cosmic-comp.ron"));
 
-        for path in locations {
-            debug!("Trying config location: {}", path.display());
-            if path.exists() {
-                info!("Using config at {}", path.display());
-                let Ok(file) = OpenOptions::new().read(true).open(path) else {
-                    error!("Failed to open config file.");
-                    continue;
-                };
-                match ron::de::from_reader::<_, StaticConfig>(file) {
-                    Ok(mut config) => {
-                        key_bindings::add_default_bindings(
-                            &mut config.key_bindings,
-                            workspace_layout,
-                        );
-                        return config;
-                    }
-                    Err(err) => {
-                        error!("Malformed config file (skipping): {}", err);
-                        continue;
+        // Source key bindings from com.system76.CosmicSettings.Shortcuts
+        let settings_context = shortcuts::context().unwrap();
+        let system_actions = shortcuts::system_actions(&config);
+        let mut shortcuts = shortcuts::shortcuts(&settings_context);
+
+        // Add any missing default shortcuts recommended by the compositor.
+        key_bindings::add_default_bindings(&mut shortcuts, workspace.workspace_layout);
```

Comment:
Could this override shortcuts explicitly set by the user? If so, should we not just apply these to some kind of default config, before merging the configuration? Though a merge might also be complicated, if the user wants to unbind default shortcuts. Perhaps the direction-specific shortcuts need some kind of `Auto`-variant for the default behavior?

Reply:
This function will avoid inserting bindings that have already been set, so it won't override settings from the user. I'm open to ideas for moving this into the defaults config.